language-server/language-server-api/src/main/scala/aqua/lsp/LspSemantics.scala

```diff
@@ -67,21 +67,18 @@ class LspSemantics[S[_]] extends Semantics[S, LspContext[S]] {
     RawSemantics
       .interpret(ast, initState, init.raw)
       .map { case (state, ctx) =>
-        NonEmptyChain
-          .fromChain(state.errors)
-          .fold[ValidatedNec[SemanticError[S], LspContext[S]]] {
-            Valid(
-              LspContext(
-                raw = ctx,
-                rootArrows = state.names.rootArrows,
-                constants = state.names.constants,
-                abDefinitions = state.abilities.definitions,
-                locations = state.locations.allLocations,
-                importTokens = importTokens,
-                tokens = state.locations.tokens
-              )
-            )
-          }(Invalid(_))
+        Valid(
```

Comment:
I don't like having `Validated[SemanticError, LspContext]` return type when `LspContext` has errors in it and we don't return `Invalid` in any case.
Maybe `RawSemantics` and `LspSemantics` should not extend the same `Semantic` trait then?